### DISCUSSION TITLE
Control Refs declarations

### DISCUSSION THREAD

**Discussion by pamoroso** [flet-dev/flet#185] [Q&A] [answered]
In the Flet documentation, the [Getting started](https://flet.dev/docs/guides/python/getting-started) section of the Python Guide also discusses [Control Refs](https://flet.dev/docs/guides/python/getting-started#control-refs).

I understand the rationale behind Control Refs and how they are used. But, as a beginner, what a Control Ref declaration is supposed to mean in Python eludes me. For example, in this code:
```python
first_name = Ref[TextField]()
```
is the right hand side of the assignment indexing a `Ref` list and calling the resulting item? Is it a type declaration? Something else?

**Comment by FeodorFitsner** [comment, marked as answer]
`Ref` is a generic type. Take a look at its sources: https://github.com/flet-dev/flet/blob/main/sdk/python/flet/ref.py
`Ref[TextField]()` in Python means the same as `List<string>()` in C#. Well, in Python it's just for type hinting, so IDE know what members to show when you type `first_name.current.` :)

Looks like a good article: https://medium.com/@steveYeah/using-generics-in-python-99010e5056eb

**Reply by pamoroso** [reply]
Thanks, I'm not yet familiar with Python type hints.
